src/main/java/jenkins/metrics/api/Metrics.java

```diff
@@ -490,6 +485,10 @@ private void execute(TaskListener listener) throws IOException, InterruptedExcep
             SortedMap<String, HealthCheck.Result> results;
             try {
                 results = registry.runHealthChecks(threadPoolForHealthChecks);
+            } catch (RejectedExecutionException e) {
+                listener.error("Health checks execution is rejected due to: {0}", e.getMessage());
+                LOGGER.log(Level.INFO, "Health checks execution is rejected due to: {0}", e.getMessage());
```

Comment:
Previously there was no need to keep the stack trace because the reason was putting too many executions in the pool, but now it should never happen with the new handler, so I'll print the stack trace